### PR TITLE
Add one click button.

### PR DESCRIPTION
**Pull request by DenisPopov15**

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "AWS_SECRET_ACCESS_KEY": "test",
+    "FOG_DIRECTORY": "test",
+    "HOST": "test",
+    "SMTP_ADDRESS": "test",
+    "SMTP_DOMAIN": "test",
+    "SMTP_PASSWORD": "test",
+    "SMTP_USERNAME": "test"
+  },
+  "image": "heroku/ruby",
+  "addons": [
```
Comment (alexkravets) [inline app.json:26]: We need only mongolab.

Comment (alexkravets) [inline app.json:26]: Actually we also need sendgrid.

File: app.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "web platform"
+  ],
+  "website": "https://isstvo.com/",
+  "repository": "https://github.com/alexkravets/isstvo.com",
+  "logo": "https://avatars2.githubusercontent.com/u/477851?v=3&s=400",
+  "success_url": "/",
+  "scripts": {},
+  "env": {
+    "MONGODB_URI": "test",
+    "ASSET_HOST": "test",
```
Comment (alexkravets) [inline app.json]: Is there a way to auto populate this value?

File: app.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "creatives",
+    "web platform"
+  ],
+  "website": "https://isstvo.com/",
+  "repository": "https://github.com/alexkravets/isstvo.com",
+  "logo": "https://avatars2.githubusercontent.com/u/477851?v=3&s=400",
+  "success_url": "/",
+  "scripts": {},
+  "env": {
+    "MONGODB_URI": "test",
```
Comment (alexkravets) [inline app.json]: This should also take value that's added by mongolab plugin.

File: app.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    "ASSET_HOST": "test",
+    "AWS_ACCESS_KEY_ID": "test",
+    "AWS_SECRET_ACCESS_KEY": "test",
+    "FOG_DIRECTORY": "test",
+    "HOST": "test",
```
Comment (alexkravets) [inline app.json]: This should be prepopulated with appname.herokuapp.com — by default.

File: app.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    "SMTP_ADDRESS": "test",
+    "SMTP_DOMAIN": "test",
```
Comment (alexkravets) [inline app.json]: This should be heroku.com

File: app.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  "success_url": "/",
+  "scripts": {},
+  "env": {
+    "MONGODB_URI": "test",
+    "ASSET_HOST": "test",
+    "AWS_ACCESS_KEY_ID": "test",
+    "AWS_SECRET_ACCESS_KEY": "test",
+    "FOG_DIRECTORY": "test",
+    "HOST": "test",
+    "SMTP_ADDRESS": "test",
```
Comment (alexkravets) [inline app.json]: This and username/password should be taken from variables added by sendgrid.

**Comment by DenisPopov15** [comment]
@alexkravets Yep, I am working now on fetching variables on fly to env.vars (there are now direct way how to do it), but I am sure, that this can be implemented.

**Comment by DenisPopov15** [comment]
@alexkravets updated.

File: README.md [hunk 7]
```diff
@@ -29,20 +29,8 @@ is [purposefully excluded from the project's `Gemfile`][exclude].
...
-    AWS_SECRET_ACCESS_KEY
-    FOG_DIRECTORY
-    HOST
-    SMTP_ADDRESS
-    SMTP_DOMAIN
-    SMTP_PASSWORD
-    SMTP_USERNAME
+[![Deploy](https://www.herokucdn.com/deploy/button.svg)](https://heroku.com/deploy)
 
+After deploy run `rake heroku:after_deploy` to setup env. variables and s3[optional]
```
Comment (alexkravets) [inline README.md:34]: Please add scenario here on how to deploy to heroku. How/when local app is connected to heroku app? Or there is no local app? We need step by step instructions here.

**Comment by alexkravets** [comment]
Also let's split one big take tasks into two tasks under deploy folder. The first one is env configuration for heroku deployment and second one is adding s3 bucket and s3 variables to env. This way we'll have s3 as optional thing so you don't run it if you don't need it.

**Comment by alexkravets** [comment]
Please describe all requirements for s3 deployment in readme file. It should have information about all requirements and where to get them to have s3 working from scratch.

**Pull request closed without merging**